iceoryx_hoofs/include/iceoryx_hoofs/cxx/functional_interface.hpp

```diff
@@ -117,28 +117,32 @@ struct AndThenWithValue
     ///        happens.
     /// @param[in] callable will be called when valid
     /// @return reference to *this
-    Derived& and_then(const and_then_callback_t& callable) & noexcept;
+    template <typename Functor>
```

Comment:
Please add an `enable_if<cxx::is_invocable<Functor>` narrow down the set of candidates during overload resolution.

Reply:
A comment mentioning why this was templatized would be much appreciated :blush:

Reply:
`enable_if` is ugly and has the effect that the user get even uglier template error messages - we should avoid this.

I added a `static_assert<cxx::is_invocable...` which has the same effect but in readable.